src/main/cli/cli.c

```diff
@@ -1314,17 +1315,26 @@ static void cliSerial(const char *cmdName, char *cmdline)
     memset(&portConfig, 0 , sizeof(portConfig));
 
     uint8_t validArgumentCount = 0;
+    int val = 0;
 
     const char *ptr = cmdline;
+    char *tok = cmdline;
+    serialPortIdentifier_e identifier = findSerialPortByName(strsep(&tok, " "), strcasecmp);
+    if (identifier == SERIAL_PORT_NONE) {
+        identifier = atoi(ptr++);
+    } else {
+        ptr = tok;
+    }
```

Comment:
- it is important to check parsing in integer case, otherwise `invalid` will be accepted as UART1
- convert integer to correct uart from old scleme (0 -> uart1)   
- ( ptr++ would point to '\0' writen by `strsep` )
```suggestion
    char *tok = strsep(&ptr " ");
    // try port name first
    serialPortIdentifier_e identifier = findSerialPortByName(tok, NULL);
    if (identifier == SERIAL_PORT_NONE) {
        // try integer format next 
        char* eptr;
        identifier = strtoul(tok, &eptr, 10);
        if (*eptr)                        // parsing ended before end of token - refuse garbage in identifier
            identifier = SERIAL_PORT_NONE;
        } else {
           // with integer port specification, UART1 == 0
           if (identifier >= 0 && identifier < SERIAL_UART_MAX - 1 + SERIAL_UART_FIRST_INDEX) { // adjust uarts only, handle UART0 correctly
               identifier += SERIAL_PORT_UART1;
           }
       }
   }
   
    
```

Reply:
Should be sorted - however I have not been able to test (travelling)